src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1569,6 +1570,34 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Gets the request body decoded as array, typically from a JSON payload.
+     *
+     * @throws JsonException When the body cannot be decoded to an array
+     */
+    public function toArray(): array
+    {
+        if ('' === $content = $this->getContent()) {
+            throw new JsonException('Response body is empty.');
+        }
+
+        try {
+            $content = json_decode($content, true, 512, \JSON_BIGINT_AS_STRING | (\PHP_VERSION_ID >= 70300 ? \JSON_THROW_ON_ERROR : 0));
+        } catch (\JsonException $e) {
+            throw new JsonException('Could not decode request body.', $e->getCode(), $e);
+        }
+
+        if (\PHP_VERSION_ID < 70300 && \JSON_ERROR_NONE !== json_last_error()) {
+            throw new JsonException(json_last_error_msg(), json_last_error());
```

Comment:
For PHP < 7.3 we get a different error message `Syntax error` for the same invalid JSON content.

I'd suggest keeping the same error message for consistency: `Could not decode request body.` and maybe add `\JsonException(json_last_error_msg())` as previous exception?

Reply:
Actually. Most of this function is copied from HttpClient. See

https://github.com/symfony/symfony/blob/v5.1.5/src/Symfony/Component/HttpClient/Response/ResponseTrait.php#L137-L172

I'll make sure to rewrite the test for PHP < 7.3

Reply:
I made the exception message for 7.2 like: 

> Could not decode request body: Syntax error